app/api/originalURLHandler.js

```diff
@@ -1,35 +1,37 @@
-var MongoClient = require('mongodb').MongoClient;
 var winston = require('winston');
 
 var findURL = require('./databaseInteraction/findURL.js');
 var insertURL = require('./databaseInteraction/insertURL.js');
 
-module.exports = function(paramUrl, dbURL, callback) {
-	MongoClient.connect(dbURL, function(e, db) {
-		if (e) { winston.log('error', e); }
-		findURL(db,'original_url', paramUrl, function(result) {
-			// if we find a match in the database, just return the match
-			if (result) {
+module.exports = function(paramUrl, dbURL) {
+	return new Promise(function(resolve, reject) {
+		// Open the database and search for the original URL
+		findURL(dbURL, 'original_url', paramUrl)
+		// If there are no matches, insert a new document
+		.then(function(result) {
+			if (result == null) {
+				handleNewURL();
+			// If there is a match, log it, then return it to index.js
+			} else {
 				winston.log('info', 'URL already exists in the database: ' + JSON.stringify({
 					original_url: result.original_url,
 					short_url: result.short_url
 				}));
-				db.close();
-				winston.log('info', 'database closed');
-				callback('URLExisted', result);
-			// if the url wasn't in the database, insert it
-			} else {
-				winston.log('info', 'New URL being send to insertURL');
-				insertURL(db, paramUrl, function(result) {
-					winston.log('info', 'New data inserted into database: \n' + JSON.stringify({
-						original_url: result.ops[0].original_url,
-						short_url: result.ops[0].short_url
-					}));
-					db.close();
-					winston.log('info', 'database closed');
-					callback('URLNew', result);
-				});
+				resolve(result);
 			}
+		}).catch(function(reason) {
+			reject(reason);
```

Comment:
oh yeah totally.  That error _should_ go away once you just return this whole promise instead of wrapping it in a `new Promise`. Then your `catch` in app/routes/index.js will catch it I believe